CONTRIBUTING.md

```diff
@@ -1,47 +1,224 @@
+    - Try to maintain a single feature/bugfix per pull request. It's okay to introduce a little bit of housekeeping changes along the way, but try to avoid conflating multiple features. Eventually all these are going to go into a single commit, so you can use that to frame your scope.
+
+### Step 5: Commit
+
+Create a commit with the proposed change changes:
+
+- Commit message should describe motivation. Think about your code reviewers and what information they need in order to understand what you did. If it's a big commit (hopefully not), try to provide some good entry points so it will be easier to follow.
+
+### Step 6: Pull Request
+
+- Push your changes to your GitHub fork
+- Submit a Pull Requests on the aws-sdk-ios repo to the `master` branch and add the Amplify Native team using @aws-amplify/amplify-native so we can approve/provide feedback.
+- The title of your PR must be descriptive to the specific change.
+- No period at the end of the title.
+- Pull Request message should indicate which issues are fixed: `fixes #<issue>` or `closes #<issue>`.
+- PR messaged should include shout out to collaborators.
+- If not obvious (i.e. from unit tests), describe how you verified that your change works.
+- If this PR includes breaking changes, they must be listed at the top of the changelog as described above in the Pull Request Checklist.
+- Discuss review comments and iterate until you get at least one “Approve”. When iterating, push new commits to the same branch. 
+- Usually all these are going to be squashed when you merge to master.
+- Make sure to update the PR title/description if things change. 
+- Rebase with master if the master branch has commits ahead of your fork.
+
+### Step 7: Merge
+Once your PR has been approved and tested, go ahead and merge it into `master`! Barring any unforeseen circumstances, your changes will be released in our next version. Yay!! 
+
+## Troubleshooting
+
+Some build issues can be solved by [removing your derived data](https://iosdevcenters.blogspot.com/2015/12/how-to-delete-derived-data-and-clean.html) and doing a clean and build. For any other serious build issues, open a new issue or see if there is one existing that may have a fix on it.
+
+## Related Repositories
+
+This project is part of the Amplify Framework, which runs on Android,
+iOS, and numerous JavaScript-based web platforms.
+
+1. [AWS Amplify for iOS](https://github.com/aws-amplify/amplify-ios)
+2. [AWS Amplify for JavaScript](https://github.com/aws-amplify/amplify-js)
+
+AWS Amplify plugins are built on top of the AWS SDKs. AWS SDKs are a
+toolkit for interacting with AWS backend resources.
+
+1. [AWS SDK for Android](https://github.com/aws-amplify/aws-sdk-android)
+2. [AWS SDK for iOS](https://github.com/aws-amplify/aws-sdk-ios)
+3. [AWS SDK for JavaScript](https://github.com/aws/aws-sdk-js)
+
+Not officially part of the AWS SDKs, [AppSync](https://aws.amazon.com/appsync/) is an opinionated,
```

Comment:
The API & DataStore plugins use AppSync service, but not the AppSync library, right? I'd prefer not to confuse matters by bringing up the library here.

Reply:
this is in the android contributing doc, so I just put it here, ill remove this seciton entirely